### DISCUSSION TITLE
Error when building in LabDemos

### DISCUSSION THREAD

**Discussion by farahshribaty** [MuhammadSulaiman001/opengl-lab#19] [Q&A] [answered]
ERROR::SHADER_COMPILATION_ERROR of type: VERTEX
ERROR: 0:1: '' :  Incorrect GLSL version: 460
WARNING: 0:2: 'GL_ARB_explicit_attrib_location' :  extension is not available in current GLSL version
WARNING: 0:2: 'GL_ARB_explicit_attrib_location' :  extension is not available in current GLSL version
WARNING: 0:2: 'GL_ARB_explicit_attrib_location' :  extension is not available in current GLSL version
ERROR: 0:10: 'projection' : undeclared identifier
ERROR: 0:10: 'view' : undeclared identifier
ERROR: 0:10: 'model' : undeclared identifier
ERROR: 0:10: 'aPos' : undeclared identifier
ERROR: 0:10: 'constructor' : not enough data provided for construction

 -- --------------------------------------------------- --
ERROR::SHADER_COMPILATION_ERROR of type: FRAGMENT
ERROR: 0:1: '' :  Incorrect GLSL version: 460
ERROR: 0:6: 'FragColor' : undeclared identifier
ERROR: 0:6: 'assign' :  cannot convert from 'const 4-component vector of float' to 'float'

 -- --------------------------------------------------- --
ERROR::PROGRAM_LINKING_ERROR of type: PROGRAM
Attached vertex shader is not compiled.

 -- --------------------------------------------------- --

**Comment by MuhammadSulaiman001** [comment, marked as answer]
Hi @farahshribaty, try changing `#version 460 core` to `#version 330 core` in all [vs](https://github.com/MuhammadSulaiman001/opengl-lab/tree/master/OpenGLab/LabDemos/shaders/fs) and [fs](https://github.com/MuhammadSulaiman001/opengl-lab/tree/master/OpenGLab/LabDemos/shaders/fs) files, then rebuild `LabDemos` project.

If this does not work, please provide the full logs from output window (Visual studio -> View -> Output).

**Reply by farahshribaty** [reply]
it workss thank you..
